packages/next/src/server/dev/turbopack-utils.ts

```diff
@@ -179,3 +191,39 @@ export async function getTurbopackJsConfig(
   const { jsConfig } = await loadJsConfig(dir, nextConfig)
   return jsConfig ?? { compilerOptions: {} }
 }
+
+export async function readPartialManifest<T>(
+  distDir: string,
+  name:
+    | typeof MIDDLEWARE_MANIFEST
+    | typeof BUILD_MANIFEST
+    | typeof APP_BUILD_MANIFEST
+    | typeof PAGES_MANIFEST
+    | typeof APP_PATHS_MANIFEST
+    | `${typeof SERVER_REFERENCE_MANIFEST}.json`
+    | `${typeof NEXT_FONT_MANIFEST}.json`
+    | typeof REACT_LOADABLE_MANIFEST,
+  pageName: string,
+  type:
+    | 'pages'
+    | 'app'
+    | 'app-route'
+    | 'middleware'
+    | 'instrumentation' = 'pages'
+): Promise<T> {
+  const manifestPath = posix.join(
+    distDir,
+    `server`,
+    type === 'app-route' ? 'app' : type,
+    type === 'middleware' || type === 'instrumentation'
+      ? ''
+      : pageName === '/'
+      ? 'index'
+      : pageName === '/index' || pageName.startsWith('/index/')
+      ? `/index${pageName}`
+      : pageName,
```

Comment:
(needs to be imported)

```suggestion
      : getAssetPathFromRoute(pageName),
```